### PR TITLE
[math] Refactor taichi operators

### PR DESCRIPTION
**Pull request by Routhleck**
Add dnn.linear taichi implmentation

### CODE REVIEW

**Review by Routhleck — COMMENTED** [review]
Ready to pull request

File: brainpy/_src/dnn/linear.py [hunk 1]
```diff
@@ -8,6 +8,7 @@
 import jax.numpy as jnp
 import numba
 import numpy as np
+import taichi as ti
```
Comment (chaoming0625) [inline brainpy/_src/dnn/linear.py:11]: Do not use this kind of import internally.

**Review by chaoming0625 — CHANGES_REQUESTED** [review]
I am confused by the code in this PR. The discussion may be needed.

**Comment by Routhleck** [comment]
Replace with taichi op 

* [x] event csr matvec:
  cpu default: taichi
  gpu default: brainpylib(cusparse)
* [x] csr matvec
  cpu default: taichi
  gpu default: taichi
* [x] jitconn event matvec
  cpu default: taichi
  gpu default: taichi(outdim parallel=False) brainpylib(outdim parallel=True)
* [x] jitconn matvec
  cpu default: taichi
  gpu default: taichi(outdim parallel=False) brainpylib(outdim parallel=True)

**Comment by Routhleck** [comment]
Set all operators to using taichi kernel defaultly

**Review by chaoming0625 — APPROVED** [review]
Great!

**Pull request merged**